packages/model-fragments/lib/fragments/model.js

```diff
@@ -241,5 +135,29 @@ function getActualFragmentType(declaredType, options, data) {
   return actualType || declaredType;
 }
 
+// Returns the internal model for the given record/fragment
+export function internalModelFor(record) {
+  var internalModel = record._internalModel;
+
+  // Ensure the internal model has a fragments hash, since we can't override the
+  // constructor function anymore
+  if (!internalModel._fragments) {
+    internalModel._fragments = create(null);
+  }
+
+  return internalModel;
+}
```

Comment:
Regarding this, the pattern I see emerging is people doing:

```
const keys = Object.keys || Ember.keys;
const create = Object.create || Ember.create;
```

That way, only terrible browsers get the deprecation warning. I'm not sure what the plan is regarding sharing these shims among packages. I don't know why Ember doesn't want to expose the shim themselves since it has to have the shim anyway.